src/Apps/Legion/Components/CookSource/CookSource.tsx

```diff
@@ -0,0 +1,284 @@
+            onSourceChange?.(newSource);
+            onGetTableData?.(null);
+        },
+        [onGetTableData, onSourceChange, selectedSource]
+    );
+    const handleSourceDatabaseChange = useCallback(
+        (databaseName: string) => {
+            const newSource: IADXConnection = {
+                ...(selectedSource as IADXConnection),
+                database: databaseName,
+                table: null,
+                twinIdColumn: null,
+                tableData: null
+            };
+            setSelectedSource(newSource);
+            onSourceChange?.(newSource);
+            onGetTableData?.(null);
+        },
+        [onGetTableData, onSourceChange, selectedSource]
+    );
+    const handleSourceTableChange = useCallback(
+        (tableName: string) => {
+            const newSource: IADXConnection = {
+                ...(selectedSource as IADXConnection),
+                table: tableName,
+                twinIdColumn: null,
+                tableData: null
+            };
+            setSelectedSource(newSource);
+            onSourceChange?.(newSource);
+            onGetTableData?.(null);
+
+            getTableState.callAdapter({
+                databaseName: (selectedSource as IADXConnection).database,
+                tableName: tableName
+            });
+        },
+        [getTableState, onGetTableData, onSourceChange, selectedSource]
+    );
+    const handleSourceTwinIDColumnChange = useCallback(
+        (_event, option: IDropdownOption) => {
+            const newSource: IADXConnection = {
+                ...(selectedSource as IADXConnection),
+                twinIdColumn: option.text
+            };
+            setSelectedSource(newSource);
+            onSourceChange?.(newSource);
+        },
+        [onSourceChange, selectedSource]
+    );
+
+    const handlePIDUrlChange = useCallback(
+        (_event: FormEvent<HTMLDivElement>, option?: IDropdownOption<any>) => {
+            const newSource: IPIDDocument = {
+                ...(selectedSource as IPIDDocument),
+                pidUrl: option.text
+            };
+            setSelectedSource(newSource);
+            onSourceChange?.(newSource);
+        },
+        [onSourceChange, selectedSource]
+    );
+
+    // side effects
+    useEffect(() => {
+        if (getTableState?.adapterResult?.result) {
+            const data = getTableState.adapterResult.getData() as ITable;
+            setSelectedSource(
+                produce((draft: WritableDraft<IADXConnection>) => {
+                    draft.tableData = data;
+                    draft.tableType = getTableSchemaTypeFromTable(data);
+                })
+            );
+            setSourceTableColumnOptions(
+                data.Columns.map((d) => ({
+                    key: d.columnName,
+                    text: d.columnName
+                }))
+            );
+            onGetTableData?.(data);
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
why do we need to disable deps here?

Reply:
I only want to use side effects with only that change in the dependency list

Reply:
won't that mean that when propB changes, you might have a stale reference to propA that isn't in the list? maybe not since it isn't a callback? 🤔

Reply:
okay, it seems like it is fine here, removing the disabling